cmake/modules/SIMDExt.cmake

```diff
@@ -16,49 +16,21 @@
 
 if(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64|AARCH64")
   set(HAVE_ARM 1) 
-  set(save_quiet ${CMAKE_REQUIRED_QUIET})
-  set(CMAKE_REQUIRED_QUIET true)
-  include(CheckCXXSourceCompiles)
-
-  check_cxx_source_compiles("
-    #define CRC32CX(crc, value) __asm__(\"crc32cx %w[c], %w[c], %x[v]\":[c]\"+r\"(crc):[v]\"r\"(value))
-    asm(\".arch_extension crc\");
-    unsigned int foo(unsigned int ret) {
-      CRC32CX(ret, 0);
-      return ret;
-    }
-    int main() { foo(0); }" HAVE_ARMV8_CRC)
-    check_cxx_source_compiles("
-    asm(\".arch_extension crypto\");
-    unsigned int foo(unsigned int ret) {
-      __asm__(\"pmull  v2.1q,          v2.1d,  v1.1d\");
-      return ret;
-    }
-    int main() { foo(0); }" HAVE_ARMV8_CRYPTO)
-
-  set(CMAKE_REQUIRED_QUIET ${save_quiet})
-  if(HAVE_ARMV8_CRC)
-    message(STATUS " aarch64 crc extensions supported")
-  endif()
-
-  if(HAVE_ARMV8_CRYPTO)
-    message(STATUS " aarch64 crypto extensions supported")
-  endif()
-  CHECK_C_COMPILER_FLAG(-march=armv8-a+crc+crypto HAVE_ARMV8_CRC_CRYPTO_MARCH)
-
-  # don't believe only the -march support; gcc 4.8.5 on RHEL/CentOS says
-  # it supports +crc but hasn't got the intrinsics or arm_acle.h.  Test for
-  # the actual presence of one of the intrinsic functions.
-  if(HAVE_ARMV8_CRC_CRYPTO_MARCH)
-    check_cxx_source_compiles("
-      #include <inttypes.h>
-      int main() { uint32_t a; uint8_t b; __builtin_aarch64_crc32b(a, b); }
-    " HAVE_ARMV8_CRC_CRYPTO_INTRINSICS)
-  endif()
+  include(CheckCCompilerFlag)
 
+  check_c_compiler_flag(-march=armv8-a+crc+crypto HAVE_ARMV8_CRC_CRYPTO_INTRINSICS)
   if(HAVE_ARMV8_CRC_CRYPTO_INTRINSICS)
-    message(STATUS " aarch64 crc+crypto intrinsics supported")
-    set(ARMV8_CRC_COMPILE_FLAGS "${ARMV8_CRC_COMPILE_FLAGS} -march=armv8-a+crc+crypto")
+    set(ARMV8_CRC_COMPILE_FLAGS "-march=armv8-a+crc+crypto")
+    set(HAVE_ARMV8_CRC TRUE)
+    set(HAVE_ARMV8_CRYPTO TRUE)
+  else()
+    check_c_compiler_flag(-march=armv8-a+crc HAVE_ARMV8_CRC)
+    check_c_compiler_flag(-march=armv8-a+crypto HAVE_ARMV8_CRYPTO)
```

Comment:
Is there any compiler that only has one out of them and is supported to compile ceph?

Reply:
@agraf TBH, i don't know. probably it depends on how a GCC was configured by its packager. but after reading through https://gcc.gnu.org/install/configure.html, i still cannot figure out how we can specify the supported mcpu extension set supported by the built GCC. 

in this change, i am just trying to follow how we check the CPU extension before this change. if you think we can assume that we can always have `crc` both `crypto` or neither of them, i am happy to drop the stuff in `else` branch.

Reply:
I don't know for sure either, but it seems a bit clunky to support every combination of things there is. Can you check with someone who knows gcc better than me?

Reply:
> support every combination of things there is

@agraf we can just have loop like

```
foreach(extension crc crypto)
endforeach()
```

Reply:
@agraf thanks for your review. i will ping Dan on this also to see if he can help check this.